dex/order/order.go

```diff
@@ -610,7 +610,7 @@ func ValidateOrder(ord Order, status OrderStatus, lotSize uint64) error {
 		// Market orders OK statuses: epoch and executed (NOT booked or
 		// canceled).
 		switch status {
-		case OrderStatusEpoch, OrderStatusExecuted:
+		case OrderStatusEpoch, OrderStatusExecuted, OrderStatusRevoked:
```

Comment:
Both market and limit orders can be given revoked status. Note that this involves a somewhat strange status transition for market orders from executed to revoked (both in the archived/inactive orders table for a market).